operators/pkg/instance-controller/virtualmachines_test.go

```diff
@@ -122,9 +122,34 @@ var _ = Describe("Generation of the virtual machine and virtual machine instance
 		err = reconciler.EnforceVMEnvironment(ctx)
 	})
 
-	It("Should enforce the cloud-init secret", func() {
-		// Here, we only check the secret presence to assert the function execution, leaving the other assertions to the proper tests.
-		Expect(reconciler.Get(ctx, objectName, &secret)).To(Succeed())
+	Context("The environment mode is Standard", func() {
+		BeforeEach(func() {
+			environment.Mode = clv1alpha2.ModeStandard
+		})
+		It("Should enforce the cloud-init secret", func() {
+			// Here, we only check the secret presence to assert the function execution, leaving the other assertions to the proper tests.
+			Expect(reconciler.Get(ctx, objectName, &secret)).To(Succeed())
+		})
+	})
+
+	Context("The environment mode is Exam", func() {
+		BeforeEach(func() {
+			environment.Mode = clv1alpha2.ModeExam
+		})
+		It("Should not enforce the cloud-init secret", func() {
+			// Here, we only check the secret presence to assert the function execution, leaving the other assertions to the proper tests.
+			Expect(reconciler.Get(ctx, objectName, &secret)).To(Not(Succeed()))
```

Comment:
Nit. Should update the comment, and would check the error is actually not found as [here](https://github.com/netgroup-polito/CrownLabs/blob/4638f801179722dac3c45eb3d2e40b152b51041b/operators/pkg/instance-controller/virtualmachines_test.go#L192).